daemon/algod/api/server/v2/handlers.go

```diff
@@ -1176,25 +1176,35 @@ func (v2 *Handlers) GetApplicationBoxes(ctx echo.Context, applicationID uint64,
 }
 
 // GetApplicationBoxByName returns the value of an application's box
-// (GET /v2/applications/{application-id}/boxes/{box-name})
-func (v2 *Handlers) GetApplicationBoxByName(ctx echo.Context, applicationID uint64, boxName string) error {
+// (GET /v2/applications/{application-id}/box)
+func (v2 *Handlers) GetApplicationBoxByName(ctx echo.Context, applicationID uint64, params generated.GetApplicationBoxByNameParams) error {
 	appIdx := basics.AppIndex(applicationID)
 	ledger := v2.Node.LedgerForAPI()
-
 	lastRound := ledger.Latest()
 
-	boxName, err := url.PathUnescape(boxName)
+	encodedBoxName := params.Name
+	boxNameBytes, err := logic.NewAppCallBytes(encodedBoxName)
 	if err != nil {
 		return badRequest(ctx, err, err.Error(), v2.Log)
 	}
-	value, err := ledger.LookupKv(lastRound, logic.MakeBoxKey(appIdx, boxName))
+	escapedBoxName, err := url.PathUnescape(string(boxNameBytes.Value))
```

Comment:
@algochoi A couple of questions here:

* Can you help me understand why a subset of `params.Name` (`boxNameBytes.Value`) is URL decoded rather than the entire value?
* My intuition is either (a) entire query string value must be URL decoded or (b) no URL decoding needed.
  * Based on source code inspection, I think it's (b).  Rationale:
    * `daemon/algod/api/server/v2/generated/routes.go::GetApplicationBoxByName` references `ctx.QueryParams()` to access the underlying HTTP request value.
    * `QueryParams()` implementation:  https://github.com/labstack/echo/blob/v4.1.17/context.go#L331-L336.  It relies on `net.url.ParseQuery`.
    * Based on `ParseQuery` docs (https://pkg.go.dev/net/url#ParseQuery) and source code inspection, I see it performs URL decoding.

Reply:
I think you are right, it looks like the underlying `ParseQuery` will do the correct parsing - I'll delete the `PathUnescape` part

Reply:
@algochoi Thanks for the changes in https://github.com/algorand/go-algorand/pull/4183/commits/9cf837c089e0f007a26fc29a9dbc2848d2796379 - Resolving.